projects/dataset_recorder/data_recorder.cc

```diff
@@ -0,0 +1,246 @@
+// standard includes
+#include<string>
+#include<vector>
+#include <thread>
+#include <mutex>
+#include <atomic>
+
+// BoB includes
+#include "common/main.h"
+#include "common/stopwatch.h"
+#include "common/background_exception_catcher.h"
+#include "common/gps.h"
+#include "common/map_coordinate.h"
+#include "common/bn055_imu.h"
+
+#include "imgproc/opencv_unwrap_360.h"
+#include "video/panoramic.h"
+
+#include <sys/stat.h>
+#include <ctime>
+
+
+
+
+
+
+// Standard C++ includes
+#include <iostream>
+
+using namespace BoBRobotics;
+using namespace BoBRobotics::ImgProc;
+using namespace BoBRobotics::Video;
+
+
+#define WIDTH 192
```

Comment:
``constexpr int width = 192;`` is much nicer

Reply:
okay, but what is the reason? Compiler likes it more?

Reply:
It's more modern. That way your constant has a type and a namespace etc. and it won't clash with some value in a C header that's also called ``WIDTH``.

And the type thing can be useful for finding errors at compile time: e.g. with your ``UPDATE_INTERVAL``, if you do it as I suggested below then it'll have a type of ``std::chrono::milliseconds`` which will stop you from accidentally using it as a value in seconds.